src/main/scala/com/chaordicsystems/cactus/Parser.scala

```diff
@@ -0,0 +1,60 @@
+package com.chaordicsystems.cactus
+
+import com.chaordicsystems.cactus.Operator._
+import com.chaordicsystems.cactus.Validator._
+import com.sksamuel.elastic4s.ElasticDsl.{bool, must, should}
+import com.sksamuel.elastic4s._
+import org.json4s._
+import org.json4s.jackson.JsonMethods._
+
+object Parser {
+  implicit val formats = DefaultFormats
+
+  def AND(args: List[QueryDefinition]): QueryDefinition = bool { must(args) }
+
+  def OR(args: List[QueryDefinition]): QueryDefinition = bool { should(args) minimumShouldMatch 1 }
+
+  def handleUnary(operation: JValue, typeEnabled: Boolean): QueryDefinition = {
+    val (op, field, args) = validateUnaryOperation(operation)
+    op match {
+      case Operator.NE => field NE (args, typeEnabled)
+      case Operator.EQ => field EQ (args, typeEnabled)
+      case Operator.LT => field LT (args, typeEnabled)
+      case Operator.GT => field GT (args, typeEnabled)
+      case Operator.LE => field LE (args, typeEnabled)
+      case Operator.GE => field GE (args, typeEnabled)
+    }
+  }
+
+  def handleBinary(operation: JValue, typeEnabled: Boolean): QueryDefinition = {
+    val op = validateOperator(operation)
+    val args = validateBinaryArgs(operation)
+    op match {
+      case Operator.AND => AND(args.map(validateAndTranslate(_, typeEnabled)))
+      case Operator.OR => OR(args.map(validateAndTranslate(_, typeEnabled)))
+    }
+  }
+
+  def handleMultiary(operation: JValue): QueryDefinition = {
+    val (op, field, args) = validateMultiaryOperation(operation)
+    op match {
+      case Operator.ALL => field ALL args
+      case Operator.ANY => field ANY args
+    }
+  }
+
+  def validateAndTranslate(operation: JValue, typeEnabled: Boolean): QueryDefinition = {
+    val op = validateOperator(operation)
+    if (isBinary(op)) {
+      handleBinary(operation, typeEnabled)
+    } else if (isUnary(op)) {
+      handleUnary(operation, typeEnabled)
+    } else if (isMultiary(op)) {
+      handleMultiary(operation)
+    } else {
+      throw InvalidCactusQueryFormatException()
```

Comment:
My smile was because you didn't provide a meaningful error message.
Think `else` must stay here. You cannot guarantee you'll never get here. Remember, software is maintained by many people, and people can change other parts of the code, and you will have problems if you are not defensive.